eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,38 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper($servicePrincipal)
+{
+    $spPassword = ""
+    if ($servicePrincipal.GetType().Name -eq 'MicrosoftGraphServicePrincipal') {
+        Write-Host "Creating password for service principal via MS Graph API"
+        # Microsoft graph objects (Az version >= 7.0.0) do not provision a secret # on creation so it must be added separately.
+        # Submitting a password credential object without specifying a password will result in one being generated on the server side.
+        $password = New-Object -TypeName "Microsoft.Azure.PowerShell.Cmdlets.Resources.MSGraph.Models.ApiV10.MicrosoftGraphPasswordCredential"
+        $password.DisplayName = "Password for $displayName"
+        $credential = Retry { New-AzADSpCredential -PasswordCredentials $password -ServicePrincipalObject $servicePrincipal }
+        $spPassword = ConvertTo-SecureString $credential.SecretText -AsPlainText -Force
+    } else {
+        # Secret property exists on PSADServicePrincipal type from AAD graph in Az # module versions < 7.0.0
+        $spPassword = $servicePrincipal.Secret
+    }
+
+    # MicrosoftGraphServicePrincipal objects have AppId, AAD graph objects have ApplicationId
+    $appId = $servicePrincipal.AppId ? $servicePrincipal.AppId : $servicePrincipal.ApplicationId
+
+    return @{
+        AppId = $appId
+        ApplicationId = $appId
+        # This is the ObjectId/OID but most return objects use .Id so keep it consistent to prevent confusion
```

Comment:
Should we consider adding an `ObjectId` member to make that clear to those who may try looking for it?

Reply:
This was my initial approach, but given that `.ID` is referenced on similar objects in other places, I think that by being more specific with the naming it implies that `.ID` is something separate from `ObjectId`. At least, this is a common confusion I have had. I think by using the "official" property name for the object id, it's simultaneously less clear but easier to reason about once you know that ID == object id and AppId == application id. I'm open to disagreement here if you feel strongly about it :)

Reply:
It's a fair point that I hadn't considered.   I'm good with keeping as-is.